python/simplesauce/options.py

```diff
@@ -43,128 +95,87 @@ def _set_default_build_name(self):
 
-    def parse_platform_name(self, platform):
-        if self.browser_name.lower() == 'safari':
-            self.set_mac_os()
-        elif 'windows' in platform.lower():
-            self.platform_name = 'Windows 10'
+    def __setattr__(self, key, value):
+        self.set_option(key, value)
+
+    def __getattr__(self, key):
+        if key is 'selenium_options':
+            if 'opts' in self.seleniumOptions.keys():
+                return self.seleniumOptions['opts']
+            else:
+                return None
+        try:
+            if key in sauce_configs.keys():
+                return self.options['sauce:options'][sauce_configs[key]]
+            elif key in w3c_configs.keys():
+                return self.options[w3c_configs[key]]
+            else:
+                raise AttributeError
+        except KeyError:
+            return None
+
+    # Sets with camelCase
+    def set_capability(self, key, value):
+        if key in sauce_configs.values():
+            self.options['sauce:options'][key] = value
+        elif key in w3c_configs.values():
+            self.options[key] = value
         else:
-            self.platform_name = platform
-
-        self.options['platformName'] = self.platform_name
-
-    def parse_options(self, options):
-
-        if type(options) == ChromeOptions:
-            self._set_defaults()
-        elif type(options) == FirefoxOptions:
-            self._set_defaults('firefox')
-        elif 'version' in options:
-            self.browser_name = options['browserName'] if options['browserName'] else 'chrome'
-            self.browser_version = options['version'] if options['version'] else 'latest'
-            self.platform_name = 'Windows 10' if not options['version'] else options['version']
-            self.options['browserName'] = self.browser_name
-            self.options['browserVersion'] = self.browser_version
-            self.options['platformName'] = self.platform_name
-        elif 'sauce:options' in options:
-            w3cSauceOptions = options
-            self.browser_name = w3cSauceOptions.get('browserName')
-            self.browser_version = w3cSauceOptions['sauce:options'].get('browserVersion')
-            self.platform_name = w3cSauceOptions['sauce:options'].get('platformName')
-
-            self.options['browserName'] = self.browser_name
-            self.options['browserName'] = self.browser_name
-            self.options['browserVersion'] = self.browser_version
-            self.options['platformName'] = self.platform_name
-
-            self.options['sauce:options']['name'] = w3cSauceOptions.get('name')
-            self.options['sauce:options']['build'] = w3cSauceOptions.get('build')
+            raise AttributeError
+
+    # Sets with snake_case
+    def set_option(self, key, value):
+        if key == 'browser_name':
+            self._set_browser_name(value)
+        elif key == 'platform_name':
+            self._set_platform_name(value)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Too many positional arguments for method call](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4711738)